homeassistant/components/jellyfin/config_flow.py

```diff
@@ -83,3 +85,52 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._reauth_input = dict(entry_data)
```

Comment:
Why do we need this?

Reply:
I'm not sure what you are referring to. If you mean the casting to a dict, that is required in order to be able to update the entry data in the next step.

Reply:
I think this is different from the typical/suggested way of doing things is why he raised it as question.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=reauth#reauthentication

notably you likely want to store the entry

Reply:
See here as example which is quite similar to what you want to achieve.
Not sure if it's possible to change the username or if that should also be removed from the schema?

Reply:
Okay, I'm not sure why entry_data is passed into async_step_reauth if we are not supposed to use it.
But anyway, I reconfigured the reauth flow to use the example code.